src/main/java/uk/gov/companieshouse/email_producer/EmailProducer.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.companieshouse.email_producer;
+
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import uk.gov.companieshouse.email_producer.factory.EmailFactory;
+import uk.gov.companieshouse.email_producer.model.Email;
+import uk.gov.companieshouse.email_producer.model.EmailData;
+import uk.gov.companieshouse.kafka.exceptions.SerializationException;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.kafka.serialization.AvroSerializer;
+
+@Service
+public class EmailProducer {
+
+    @Value("${email.producer.appId}")
+    private String appId;
+
+    private final EmailFactory emailFactory;
+    private final AvroSerializer<Email> serializer;
+    private final CHKafkaProducer chKafkaProducer;
+    private static final String EMAIL_SEND_TOPIC = "email-send";
+
+    public EmailProducer(final EmailFactory emailFactory, final AvroSerializer<Email> serializer,
+            final CHKafkaProducer chKafkaProducer) {
+        this.emailFactory = emailFactory;
+        this.serializer = serializer;
+        this.chKafkaProducer = chKafkaProducer;
+        System.out.println(String.format("App ID: %s", appId));
```

Comment:
Proper java logger should be used for it.

Reply:
Beside that since 

```
@Value("${email.producer.appId}")
     private String appId;
```

value will be null inside constructor and it will be set later after instance is already instantiated.

I'd probably add app ID injection to constructor.

Reply:
The log was just for debugging purposes, I'll be getting rid of it once the thing works :) 

Ahhh I finally understand how the @Qualifier annotation fits into this. Got it working!